File: java/src/main/java/dev/spiralmoon/maplestory/api/dto/StarforceHistoryResponseDTO.java
package dev.spiralmoon.maplestory.api.dto;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.ToString;

import java.util.List;

/**
 * 스타포스 히스토리 응답 정보
 */
@AllArgsConstructor
@Data
@ToString
public class StarforceHistoryResponseDTO {

    /**
     * 결과 건 수
     */
    @SerializedName("count")
    private int count;

    /**
     * 스타포스 히스토리
     */
    @SerializedName("starforce_history")
    private List<StarforceHistoryDTO> starforceHistory;

    /**
     * 페이징 처리를 위한 cursor
     */
    @SerializedName("next_cursor")
    private String nextCursor;
}

File: js/src/maplestory/api/dto/history/starforceHistoryResponseDto.ts
import { StarforceHistoryDto } from '@src/maplestory/api/dto/history/starforceHistoryDto';
import { StarforceHistoryResponseDtoBody } from '@src/maplestory/api/response/history/starforceHistoryResponseDtoBody';

/**
 * 스타포스 히스토리 응답 정보
 */
class StarforceHistoryResponseDto {
  /**
   * 결과 건 수
   */
  count: number;

  /**
   * 스타포스 히스토리
   */
  starforceHistory: StarforceHistoryDto[];

  /**
   * 페이징 처리를 위한 cursor
   */
  nextCursor: string | null;

  constructor(obj: StarforceHistoryResponseDtoBody) {
    const { count, starforce_history, next_cursor } = obj;

    this.count = count;
    this.starforceHistory = starforce_history.map(
      (origin) => new StarforceHistoryDto(origin),
    );
    this.nextCursor = next_cursor;
  }
}

export { StarforceHistoryResponseDto };
